libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java

```diff
@@ -754,27 +755,37 @@ private void updateDataFromInstruction(InstructionModel model) {
    * Looks to see if we have a new instruction text.
    * Sets new instruction text if found.
    */
-  private void updateDataFromBannerText(BannerText primaryBannerText, BannerText secondaryBannerText) {
-    if (primaryBannerText != null) {
-      InstructionLoader instructionLoader = createInstructionLoader(upcomingPrimaryText, primaryBannerText);
-      if (instructionLoader != null) {
-        instructionLoader.loadInstruction();
-      }
+  private void updateDataFromBannerText(@NonNull BannerText primaryBannerText, BannerText secondaryBannerText) {
+    if (secondaryBannerText == null) {
+      loadPrimary(primaryBannerText);
+      return;
     }
-    if (secondaryBannerText != null) {
-      if (upcomingSecondaryText.getVisibility() == GONE) {
-        upcomingSecondaryText.setVisibility(VISIBLE);
-        upcomingPrimaryText.setMaxLines(1);
-        adjustBannerTextVerticalBias(0.65f);
-      }
-      InstructionLoader instructionLoader = createInstructionLoader(upcomingSecondaryText, secondaryBannerText);
-      if (instructionLoader != null) {
-        instructionLoader.loadInstruction();
-      }
-    } else {
-      upcomingPrimaryText.setMaxLines(2);
-      upcomingSecondaryText.setVisibility(GONE);
-      adjustBannerTextVerticalBias(0.5f);
+    loadPrimaryAndSecondary(primaryBannerText, secondaryBannerText);
+  }
+
+  private void loadPrimary(BannerText primaryBannerText) {
+    upcomingPrimaryText.setMaxLines(2);
+    upcomingSecondaryText.setVisibility(GONE);
+    adjustBannerTextVerticalBias(0.5f);
+
+    InstructionLoader instructionLoader = createInstructionLoader(upcomingPrimaryText, primaryBannerText);
```

Comment:
What about extracting this block of code into a new method and reuse it here and also in `loadPrimaryAndSecondary` method below?